src/workflow/snakemake/snakemake.py

```diff
@@ -165,26 +189,40 @@ def _write_includes(f: TextIO, includes: List[str]):
         for include in includes:
             f.write(f'include: "{os.path.join(includes_path, include)}"\n')
 
-        f.write('\n')
+        f.write("\n")
 
     @staticmethod
-    def _prepare_argv(snakefile: str, cores: int, dryrun: bool, work_dir: str, **snakemake_kwargs):
+    def _prepare_argv(
+        snakefile: str,
+        cores: int,
+        dryrun: bool,
+        work_dir: str,
+        input_dir: Optional[str] = None,
+        dataset: Optional[str] = None,
+        **snakemake_kwargs,
+    ):
         """Prepare arguments to input to the snakemake cli"""
 
         argv = [
```

Comment:
This might need explicit `--use-singularity` and `--use-conda` flags sometime in the future, depending on the software management strategy

Reply:
Indeed, the flags will be added when we integrate the software env.